src/styles/main.scss

```diff
@@ -1,3 +1,55 @@
+@import "./variables.scss";
+
 body {
   margin: 0;
 }
+
+.calendar {
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+  display: flex;
+  gap: $gap;
+  flex-wrap: wrap;
+  padding: $padding;
+  width: $calendar-size;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    background-color: #eee;
+    height: $cell-size;
+    width: $cell-size;
+    border: $border solid #000;
+    align-items: center;
+    justify-content: center;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+
+      &:hover {
+        cursor: pointer;
+        background-color: pink;
+        transition: transform 0.5s;
```

Comment:
move transition from `&:hover` just to `&__day` to have smooth unhover
Plus add transition properties to specify what you'd like to change